pkg/router/template/router.go

```diff
@@ -99,7 +100,9 @@ type templateData struct {
 	// the directory that files will be written to, defaults to /var/lib/containers/router
 	WorkingDir string
 	// the routes
-	State map[string]ServiceUnit
+	State map[string](ServiceAliasConfig)
+	// the service lookup
+	ServiceUnits map[string]ServiceUnit
```

Comment:
It does make it harder, but we need two maps because now its now a many-many relationship between routes and services. A route can point to many services and a service can join many routes.
Two maps is the cleanest way, or we will end up designing a custom hybrid hash+graph. How about we revisit this after the PR? Second round of restructuring probably.